pkg/daemon/rpm-ostree.go

```diff
@@ -322,23 +322,16 @@ func (r *RpmOstreeClient) RunPivot(osImageURL string) error {
 		return errors.Wrap(err, "deleting pivot reboot-needed file")
 	}
 
-	service := "machine-config-daemon-host.service"
-	// We need to use pivot if it's there, because machine-config-daemon-host.service
-	// currently has a ConditionPathExists=!/usr/lib/systemd/system/pivot.service to
-	// avoid having *both* pivot and MCD try to update.  This code can be dropped
-	// once we don't need to care about compat with older RHCOS.
-	var err error
-	_, err = os.Stat("/usr/lib/systemd/system/pivot.service")
+	// We used to start machine-config-daemon-host here, but now we make a dynamic
+	// unit because that service was started in too many ways, and the systemd-run
+	// model of creating a unit dynamically is much clearer for what we want here;
+	// conceptually the service is just a dynamic child of this pod (if we could we'd
+	// tie the lifecycle together).  Further, let's shorten our systemd unit names
+	// by using the mco- prefix, and we also inject the RPMOSTREE_CLIENT_ID now.
+	err := exec.Command("systemd-run", "--wait", "--collect", "--unit=mco-pivot",
+		"-E", "RPMOSTREE_CLIENT_ID=mco", "/usr/libexec/machine-config-daemon", "pivot", osImageURL).Run()
```

Comment:
umm, so we are directly calling  m-c-d pivot here which means in case of an update pivot will run `systemctl reboot` (https://github.com/openshift/machine-config-operator/blob/master/cmd/machine-config-daemon/pivot.go#L253) . Can this cause system to reboot without executing remaining control flow?